src/frontend/data/stores/useLivePanelState/index.spec.ts

```diff
@@ -0,0 +1,247 @@
+    );
+    expect(useLivePanelState.getState().isPanelVisible).toBe(true);
+  });
+
+  it('updates available details and current detail selected', () => {
+    useLivePanelState.setState({
+      isPanelVisible: true,
+      currentDetail: LivePanelDetail.CHAT,
+      availableDetails: [LivePanelDetail.APPLICATION, LivePanelDetail.CHAT],
+    });
+
+    //  update with valid current selection
+    useLivePanelState
+      .getState()
+      .setAvailableDetails(
+        [LivePanelDetail.JOIN_DISCUSSION],
+        LivePanelDetail.JOIN_DISCUSSION,
+      );
+
+    expect(useLivePanelState.getState().availableDetails).toEqual([
+      LivePanelDetail.JOIN_DISCUSSION,
+    ]);
+    expect(useLivePanelState.getState().currentDetail).toEqual(
+      LivePanelDetail.JOIN_DISCUSSION,
+    );
+    expect(useLivePanelState.getState().isPanelVisible).toBe(true);
+  });
+
+  it('updates available details and select a valid detail', () => {
+    useLivePanelState.setState({
+      isPanelVisible: true,
+      currentDetail: LivePanelDetail.CHAT,
+      availableDetails: [LivePanelDetail.APPLICATION, LivePanelDetail.CHAT],
+    });
+
+    //  update with invalid current selection
+    useLivePanelState
+      .getState()
+      .setAvailableDetails(
+        [LivePanelDetail.JOIN_DISCUSSION],
+        LivePanelDetail.APPLICATION,
+      );
+
+    expect(useLivePanelState.getState().availableDetails).toEqual([
+      LivePanelDetail.JOIN_DISCUSSION,
+    ]);
+    expect(useLivePanelState.getState().currentDetail).toEqual(
+      LivePanelDetail.JOIN_DISCUSSION,
+    );
+    expect(useLivePanelState.getState().isPanelVisible).toBe(true);
+  });
+
+  it('updates available details and select one now available', () => {
+    useLivePanelState.setState({
+      isPanelVisible: true,
+      currentDetail: LivePanelDetail.CHAT,
+      availableDetails: [LivePanelDetail.APPLICATION, LivePanelDetail.CHAT],
+    });
+
+    //  restrict available details
+    useLivePanelState
+      .getState()
+      .setAvailableDetails([LivePanelDetail.APPLICATION]);
+
+    expect(useLivePanelState.getState().availableDetails).toEqual([
+      LivePanelDetail.APPLICATION,
+    ]);
+    expect(useLivePanelState.getState().currentDetail).toEqual(
+      LivePanelDetail.APPLICATION,
+    );
+    expect(useLivePanelState.getState().isPanelVisible).toBe(true);
+  });
+});
+
+describe('state.setPanelVisibility()', () => {
+  it('has no side effect', () => {
+    useLivePanelState.setState({
+      availableDetails: [LivePanelDetail.APPLICATION, LivePanelDetail.CHAT],
+      currentDetail: LivePanelDetail.CHAT,
+      isPanelVisible: true,
+    });
+
+    //  do not change the state with detail to select
+    useLivePanelState.getState().setPanelVisibility(true, LivePanelDetail.CHAT);
+
+    expect(useLivePanelState.getState().isPanelVisible).toBe(true);
+    expect(useLivePanelState.getState().currentDetail).toEqual(
+      LivePanelDetail.CHAT,
+    );
```

Comment:
Also add an assert checking that `availableDetails` has not changed. And do this assert in all other tests